File: Data Structures & Algorithms/is-anagram/submission-2.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length()){
            return false;
        }
        Map<Character,Integer> count = new HashMap<>(s.length());
        for(char c: s.toCharArray()){
            if(count.containsKey(c)){
                count.put(c,count.get(c)+1);
            }
            else{
                count.put(c,1);
            }
        }
        for(char c: t.toCharArray()){
            if(count.containsKey(c) && count.get(c)>1){
                count.put(c,count.get(c)-1);
            }
            else{
                count.remove(c);
            }
        }
        if(count.isEmpty()){
            return true;
        }
        return false;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s)==0:
            return 0
        char = set()
        l = 0
        length = 0
        for r,e in enumerate(s):
            while e in char:
                char.remove(s[l])
                l+=1
            char.add(e)
            length = max(length,r-l+1)
        return length
